tfexec/plan.go

```diff
@@ -77,8 +77,21 @@ func (opt *DestroyFlagOption) configurePlan(conf *planConfig) {
 	conf.destroy = opt.destroy
 }
 
-func (tf *Terraform) Plan(ctx context.Context, opts ...PlanOption) error {
-	return tf.runTerraformCmd(tf.planCmd(ctx, opts...))
+// Plan executes `terraform plan` with the specified options and waits for it
+// to complete.
+//
+// The returned boolean is false when the plan diff is empty (no changes) and
+// true when the plan diff is non-empty (changes present).
+//
+// The returned error is nil if `terraform plan` has been executed and exits
+// with either 0 or 2.
+func (tf *Terraform) Plan(ctx context.Context, opts ...PlanOption) (bool, error) {
+	cmd := tf.planCmd(ctx, opts...)
+	err := tf.runTerraformCmd(cmd)
+	if err != nil && cmd.ProcessState.ExitCode() == 2 {
```

Comment:
You probably need to confirm this is `ExitError`, ie something like: 
```suggestion
	var exitErr *exec.ExitError
	if errors.As(err, &exitErr) && cmd.ProcessState.ExitCode() == 2 {
```

Reply:
I don't believe this will work since the `err` here is returned from `parseError()` (hitting the `default` case). At that time the original `err` is essentially discarded.

Reply:
Should I perhaps add another case to `parseError()` that checks the original error and returns a new error type that I can then check from `Plan()`?

Reply:
yeah, i think we should probably fix the default case to just wrap the `ExitError` so that its compatible with `errors.As`/`Is`.